src/components/box/box.test.js

```diff
@@ -0,0 +1,734 @@
+        CypressMountWithProviders(
+          <BoxComponent bg="" backgroundColor={rgbValue} />
+        );
+
+        getDataElementByValue("box").should(
+          "have.css",
+          "background-color",
+          rgbValue
+        );
+      }
+    );
+
+    it.each(bgConstants)(
+      "should verify Box backgroundColor prop passed as hex value",
+      (color, rgbValue, hexValue) => {
+        CypressMountWithProviders(
+          <BoxComponent bg="" backgroundColor={hexValue} />
+        );
+
+        getDataElementByValue("box").should(
+          "have.css",
+          "background-color",
+          rgbValue
+        );
+      }
+    );
+
+    it.each(["0.1", "0.5", "1"])(
+      "should verify Box opacity is %s",
+      (opacity) => {
+        CypressMountWithProviders(
+          <BoxComponent bg="primary" opacity={opacity} />
+        );
+
+        getDataElementByValue("box").should("have.css", "opacity", opacity);
+      }
+    );
+
+    it.each(widthConstants)(
+      "should verify Box width is %s when prop is passed as a percentage",
+      (width, percentage) => {
+        CypressMountWithProviders(
+          <BoxComponent bg="primary" width={percentage} />
+        );
+
+        getDataElementByValue("box")
+          .should("have.attr", "width", percentage)
+          .and("have.css", "width", width);
+      }
+    );
+
+    it.each(widthConstants)(
+      "should verify Box width is %s when prop is passed as a number",
+      (width, percentage, number) => {
+        CypressMountWithProviders(<BoxComponent bg="primary" width={number} />);
+
+        getDataElementByValue("box")
+          .should("have.attr", "width", number)
+          .and("have.css", "width", width);
+      }
+    );
+
+    it.each(widthConstants)(
+      "should verify Box width is %s when prop is passed as a string",
+      (width) => {
+        CypressMountWithProviders(<BoxComponent bg="primary" width={width} />);
+
+        getDataElementByValue("box")
+          .should("have.attr", "width", width)
+          .and("have.css", "width", width);
+      }
+    );
+
+    it.each(heightConstants)(
+      "should verify Box height is %s when prop is passed as a number",
+      (height, number) => {
+        CypressMountWithProviders(
+          <BoxComponent bg="primary" height={number} />
+        );
+
+        getDataElementByValue("box")
+          .should("have.attr", "height", number)
+          .and("have.css", "height", height);
+      }
+    );
+
+    it.each(heightConstants)(
+      "should verify Box height is %s when prop is passed as a string",
+      (height) => {
+        CypressMountWithProviders(
+          <BoxComponent bg="primary" height={height} />
+        );
+
+        getDataElementByValue("box")
+          .should("have.attr", "height", height)
+          .and("have.css", "height", height);
+      }
+    );
+
+    it.each(sizeConstants)(
+      "should verify Box height and width are %s when Size prop is passed as a number",
```

Comment:
"should verify Box height and width are %s when **size** prop is passed as a number",